реализовала первую лабораторную+начинка проекта

NadiaDym opened the pull request.

NadiaDym committed:
реализовала первую лабораторную+начинка проекта

GMtimur commented on labs/lab1/lab1.cpp:
```diff
@@ -1,20 +1,31 @@
+    {
+        cout << "Введите число последовательности: ";
+        cin >> input;
+        if (input != 0) 
```
Вторая проверка на введенный 0 не нужна, так как уже проверяли в условии цикла while

NadiaDym replied:
Она необходима для того, чтобы 0 не заносился в стек

GMtimur commented on labs/collections/stack.h:
Нет #pragma once

NadiaDym replied:
исправила

GMtimur commented on labs/collections/stack.h:
Добавить xml документацию к типу и всему публичному контрасту

NadiaDym replied:
добавила xml и убрала комментарии

GMtimur commented on labs/collections/stack.h:
Слишком много комментариев + оформлять комментарии в соответствии с style-guide (с новой строки)

NadiaDym replied:
исправила

GMtimur commented on labs/collections/stack.h:
```diff
@@ -0,0 +1,70 @@
+		Container* next; /// ��������� �� ��������� ������� �����.
+	};
+public: /// ���������� ��������� ������ ������
+	Container* top = nullptr; /// ��������� �� ������� ������� �����, ���������������� nullptr
+	int counter = 0; /// ������� ��������� � �����, ���������������� �����.
```
Неверная инкапсуляция

NadiaDym replied:
исправила

GMtimur commented on labs/collections/stack.h:
```diff
@@ -0,0 +1,70 @@
+	{ 
+		T data; /// data: �������� ���� T, ������� �������� � �������� �����.
+		Container* next; /// ��������� �� ��������� ������� �����.
+	};
+public: /// ���������� ��������� ������ ������
```
Нужен отступ (по style-guide)

NadiaDym replied:
исправила

GMtimur commented on labs/collections/stack.h:
```diff
@@ -0,0 +1,70 @@
+template <typename T> /// ���������� �������. �������� ����� typename ������� � ���, ��� � ������� ����� �������������� ���������� ��� ������.
+class Stack /// ���������� ���������� ������ Stack, ������� ����� �������� � ����� ����� ������
```
Нужен отступ (по style-guide)

NadiaDym replied:
нет такого пункта в гайде

GMtimur commented on labs/lab1/lab1.cpp:
Нет отступов между смысловыми группами кода
<img width="583" alt="Снимок экрана 2025-02-18 в 10 16 28 PM" src="https://github.com/user-attachments/assets/f2294fb6-2785-4e9f-bb9e-24adae664e2a" />

NadiaDym replied:
исправила

GMtimur reviewed (changes requested):
На ревью тяжело прочитать комментарии, гит хаб не подгружает кодировку

NadiaDym committed:
были исправлены все ошибки

верс 2

NadiaDym committed:
исправила ошибки

версия 3

GMtimur commented on labs/collections/stack.h:
```diff
@@ -0,0 +1,101 @@
+class Stack
+{
+private:
+    struct Container
+    {
+        T data;
+        Container* next;
+    };
```
Отсутствует конструктор у структуры Container, что может привести к ошибкам при создании объектов

NadiaDym replied:
исправила

GMtimur commented on labs/collections/stack.h:
```diff
@@ -0,0 +1,101 @@
+        Container* next;
+    };
+
+    Container* top = nullptr;
+    int counter = 0;
+
+public:
```
Нужно оставить объявление переменных в приватной секции, а инициализировать в публичной секции в конструкторе класса Stack

NadiaDym replied:
исправила

NadiaDym committed:
исправила ошибки

ichepkov2401 commented on labs/collections/collections.vcxproj:
```diff
@@ -140,10 +143,11 @@
   </ItemDefinitionGroup>
   <ItemGroup>
     <ClInclude Include="framework.h" />
+    <ClInclude Include="queue.h" />
```
Откуда у вас в первой лабе очередь взялась?

ichepkov2401 replied:
Все ещё очередь в первой лабе

ichepkov2401 commented on labs/collections/collections.vcxproj.filters:
```diff
@@ -21,11 +21,14 @@
+    <ClInclude Include="stack.h">
+      <Filter>Файлы заголовков</Filter>
+    </ClInclude>
+    <ClInclude Include="queue.h">
```
Откуда у вас в первой лабе очередь взялась?

ichepkov2401 replied:
И тут тоже очередь в первой лабе

ichepkov2401 commented on labs/collections/stack.h:
```diff
@@ -0,0 +1,103 @@
+    /// </summary>
+    /// <returns>Значение верхнего элемента стека.</returns>
+    /// <exception cref="runtime_error">Выбрасывается ошибка, если стек пуст.</exception>
+    T peek()
```
Иммутабельность
```suggestion
    T peek() const
```

ichepkov2401 commented on labs/lab1/lab1.cpp:
```diff
@@ -1,20 +1,28 @@
+    setlocale(LC_ALL, "RU");
+    Stack<int> stack;
+    int input = 1;
+	while (input != 0) 
```
Табуляция поехала

NadiaDym committed:
исправила табуляцию и иммутабельность

в vs с табуляцией всё в порядке, но в гитхаб десктоп съехала на один tab еще в сторону

NadiaDym committed:
удалила очередь
